### PR TITLE
libzmqutil: add cert class and use it instead of CZMQ zcert_t

### PR DESCRIPTION
**Pull request by garlick**
This adds a `zmqutil_cert` class to replace CZMQ `zcert_t` and completes the conversion away from libczmq.

The certificate format does not change.

### CODE REVIEW

**Comment by garlick** [comment]
Fixed a TOCTOU called out by asan, and a spell error caught by the spell checker.  Yay CI!

File: src/cmd/flux-keygen.c
Comment (garlick) [reply src/cmd/flux-keygen.c:139]: Hmm, I'm not sure I agree with this one.  Basically it wants me to sanitize `path` before opening it for writing in `flux-keygen` but it sounds like they think it's a privileged command or something which it is not.  Any thoughts?

Comment (grondo) [reply src/cmd/flux-keygen.c:139]: Yeah, this seems a little silly. Will the code scanner call out every program that takes an argument to open a file? (e.g. `vi`?) Maybe @dun would have some insight, but I think this could be dismissed as a false positive.

Comment (garlick) [reply src/cmd/flux-keygen.c:139]: I wonder if it would be a good idea to define this cert format in a Flux RFC.  :thinking:

Comment (dun) [reply src/cmd/flux-keygen.c:139]: If `flux-keygen` is running as the user, I would think this would be equivalent to what they could do at the shell. So I'm not sure what the benefit would be for additional sanitization.

Comment (garlick) [reply src/cmd/flux-keygen.c:139]: Thanks @dun!  OK, I'll dismiss this one then.

File: src/common/libzmqutil/cert.c
Comment (grondo) [inline src/common/libzmqutil/cert.c:40]: A comment of where these sizes are coming from might be helpful. I'm guessing cert keys are limited to 32 bytes (256 bit) and the max expanded size of 32 bytes with z85 is 41 bytes?

**Comment by garlick** [comment]
Chasing an ASAN issue in the cert test and found some more stuff that needs testing along the way, so please hold off reviewing.

File: src/common/libzmqutil/test/cert.c
Comment (grondo) [inline src/common/libzmqutil/test/cert.c:116]: Might be good to expand the "bad" tests here, including something like a "good" cert with one character change (so should fail), some garbage input, public and/or secret key too long, unclosed quote, etc

File: scripts/install-deps-rpm.sh
Comment (grondo) [inline scripts/install-deps-rpm.sh:11]: Remove from Dockerfiles too so the next time we generate images this dependency is dropped?

Comment (grondo) [reply scripts/install-deps-rpm.sh:11]: Or is czmq still required by some other dependency?

File: src/common/libzmqutil/cert.h
Comment (grondo) [inline src/common/libzmqutil/cert.h:17]: It feels a little strange that this class is `zmqutil_cert` but the structure is just `struct cert`. However, it probably doesn't rise to the level of updating the whole PR. (just a comment)

**Review by grondo — APPROVED** [review]
This LGTM. Feel free to ignore suggestions inline, though adding the extra bad cert tests might be a good idea before merging.

**Comment by grondo** [comment]
Oops sorry @garlick! I was reviewing and missed your comment to hold off!

**Comment by garlick** [comment]
Oh that's OK,  I haven't changed that much.  Good suggestions!

File: scripts/install-deps-rpm.sh
Comment (garlick) [reply scripts/install-deps-rpm.sh:11]: Not that I know of but actually...maybe it makes sense to drop czmq in another PR just for release notes clarity?  This PR Is heavily about the cert class.

File: src/common/libzmqutil/cert.h
Comment (garlick) [reply src/common/libzmqutil/cert.h:17]: Yeah and it's a narrowly used internal class so maybe just claiming the "cert_" namespace is OK here.  I can make that change.

**Comment by garlick** [comment]
Just forced a push with a few changes
- rename cert functions from `zmqutil_cert_*()` to `cert_*()`
- add a check that secret and public keys "match" (I didn't realize there was `zmq_curve_public()`)
- fixed the ASAN issue with the test
- add more cert tests (including expanding the bad input as suggested)
- misc other minor cleanup
-  dropped the czmq build changes (will submit as  another PR)

**Comment by garlick** [comment]
@grondo -  I know you already approved but maybe you could give this one more quick look to see if anything pops out?

**Review by grondo — APPROVED** [review]
Still LGTM!

**Pull request merged**
